index.js

```diff
@@ -93,10 +93,26 @@ module.exports = {
         });
         babelPlugins.push(htmlBarsPlugin);
       }
-      this._registeredWithBabel = true;
     }
   },
 
+  /**
+   * This function checks if 'ember-cli-htmlbars-inline-precompile' is already present in babelPlugins.
+   * The plugin object will be different for non parallel API and parallel API.
+   * For parallel api, check the `baseDir` of a plugin to see if it has current dirname
+   * For non parallel api, check the 'modulePaths' to see if it contains 'ember-cli-htmlbars-inline-precompile'
+   * @param {*} plugins
+   */
+  _isBabelPluginRegistered(plugins) {
+    return plugins.some(plugin => {
```

Comment:
What is `plugins` here? In normal babel setups this would generally be an array of arrays...

Reply:
Chatted a bit offline, I think that we should do:

```js
plugins.some(plugin => {
  if (!Array.isArray(plugin)) { return false; }

  return plugin[0] === require.resolve('babel-plugin-htmlbars-inline-precompile');
});
```

And change [this line](https://github.com/ember-cli/ember-cli-htmlbars-inline-precompile/blob/8d977d84cadf3b964bf4b8cbcf257b4c4a9658cd/lib/ast-plugins.js#L5) to:

```js
const HTMLBarsInlinePrecompilePlugin = require.resolve('babel-plugin-htmlbars-inline-precompile');
```